apps/wfo-ui-surf/components/WfoServiceTicketDetailPage/WfoImpactedCustomersTable.tsx

```diff
@@ -0,0 +1,228 @@
+import React, { useContext, useState } from 'react';
+
+import { useTranslations } from 'next-intl';
+
+import {
+    EuiButtonEmpty,
+    EuiFlexGroup,
+    EuiFlexItem,
+    EuiModal,
+    EuiModalHeader,
+    EuiText,
+} from '@elastic/eui';
+import {
+    OrchestratorConfigContext,
+    WfoBadge,
+    WfoBasicTable,
+    WfoCheckmarkCircleFill,
+    WfoTableColumns,
+    WfoViewList,
+    WfoXCircleFill,
+    getStyles,
+    useFilterQueryWithRest,
+    useOrchestratorTheme,
+} from '@orchestrator-ui/orchestrator-ui-components';
+
+import { MINL_BY_SUBSCRIPTION_ENDPOINT } from '../../constants-surf';
+import { SurfConfigContext } from '../../contexts/SurfConfigContext';
+import {
+    CustomerWithContacts,
+    ImpactedCustomersTableColumns,
+    ImpactedObject,
+    ServiceTicketDefinition,
+} from '../../types';
+import { WfoImpactLevelBadge } from '../WfoBadges/WfoImpactLevelBadge';
+import { WfoCustomersContactsModal } from './WfoCustomersContactsModal';
+import { WfoImsCircuitsTable } from './WfoImsCircuitsTable';
+import { mapImpactedObjectToImpactedCustomersColumns } from './utils';
+
+interface WfoImpactedCustomersTableProps {
+    impactedObject: ImpactedObject;
+}
+
+export const WfoImpactedCustomersTable = ({
+    impactedObject,
+}: WfoImpactedCustomersTableProps) => {
+    const t = useTranslations(
+        'cim.serviceTickets.detail.tabDetails.general.subscriptionImpactCustomerTable',
+    );
+    const { cimDefaultSendingLevel } = useContext(SurfConfigContext);
+    const { theme } = useOrchestratorTheme();
+    const [contactsModal, setContactsModal] = useState<CustomerWithContacts>();
```

Comment:
I prefer verbNoun style here so isContactsModalOpen, setContactsModalOpen. Same for the next line